### PR TITLE
Document changes after separating tx messages [ECR-2730]

### PR DESCRIPTION
**Pull request by dmitry-timofeev**
Document changes after separating transactions and messages.

See also: https://github.com/exonum/exonum-java-binding/pull/574

---
Things to do:

- [x] Resolve the open questions

### CODE REVIEW

**Comment by dmitry-timofeev** [comment]
It's WIP as some questions remain (as comments in the text), but is ready to be reviewed.

File: src/get-started/java-binding.md [hunk 1]
```diff
@@ -246,16 +272,16 @@ processing on the node.
 
 The external service API is used for the interaction between the service and the
 external systems.
-A set of operations is defined by each service and can include sending
-transactions to the network, read requests for blockchain data with the
-provision of corresponding cryptographic proof, etc. Exonum provides an embedded
-web framework for implementing the REST-interface of the service.
+A set of operations defined by a service usually includes read requests
```
Comment (Maria-Nosyk) [inline src/get-started/java-binding.md]: defined by the service

File: src/get-started/java-binding.md [hunk 2]
```diff
@@ -246,16 +272,16 @@ processing on the node.
...
+for blockchain data with the provision of corresponding cryptographic proof.
```
Comment (Maria-Nosyk) [inline src/get-started/java-binding.md]: for the blockchain data with the provision of the corresponding cryptographic proof.

File: src/get-started/java-binding.md [hunk 3]
```diff
@@ -246,16 +272,16 @@ processing on the node.
 
 The external service API is used for the interaction between the service and the
 external systems.
-A set of operations is defined by each service and can include sending
-transactions to the network, read requests for blockchain data with the
-provision of corresponding cryptographic proof, etc. Exonum provides an embedded
-web framework for implementing the REST-interface of the service.
+A set of operations defined by a service usually includes read requests
+for blockchain data with the provision of corresponding cryptographic proof. 
+Exonum provides an embedded web framework for implementing the REST-interface
```
Comment (MakarovS) [inline src/get-started/java-binding.md]: `RESTful API` instead of `REST-interface`?

File: src/get-started/java-binding.md [hunk 4]
```diff
@@ -143,71 +143,90 @@ rules – see the corresponding section of our [documentation][transactions].
...
-To correctly process a transaction, it must be transformed into an
-*executable transaction* (see
-[`TransactionConverter`][transactionconvererter] or
-`Service#convertToTransaction` method) and transmitted to the framework using
-`Node#submitTransaction` method. The framework verifies it, and if the
-transactions is correct,
-broadcasts it to other nodes of the system. Other nodes, having received the
-transaction message, convert it into an executable transaction, also using the
-service transaction converter.
+1. A service client creates a transaction message, including ids
```
Comment (MakarovS) [inline src/get-started/java-binding.md]: All steps have the same number (`1`).

Comment (dmitry-timofeev) [reply src/get-started/java-binding.md]: The renderer takes care of that. It allows to reorder/remove/add things to the list easily, without rewriting all the subsequent numbers like a monkey :-)

File: src/get-started/java-binding.md [hunk 5]
```diff
@@ -222,6 +241,13 @@ reference. Light clients also provide access to information on the
 [transaction][exonum-transaction] execution result
 (which may be either success or failure) to their users.
 
+The implementation of `Transaction#execute` method must be a pure function,
+i.e. for the given transaction to produce the same _observable_ result on
+all nodes of the system. An observable result is the one that affects the
+blockchain state hash:<!-- TODO: link the definition? -->
+a write to a collection affecting the service state hash, or an execution
```
[the remark this answers is not in the document]
Comment (dmitry-timofeev) [reply src/get-started/java-binding.md]: Not sure about that — we write to collections, or we put records to collections.

File: src/get-started/java-binding.md (hunk 1 above)
Comment (dmitry-timofeev) [reply src/get-started/java-binding.md]: But I mean by _some_ (a usual) service?

File: src/get-started/java-binding.md (hunk 3 above)
Comment (dmitry-timofeev) [reply src/get-started/java-binding.md]: Yep, it is more common. Or HTTP-interface as we don't restrict anyone to follow the REST recommendations :upside_down_face:

**Comment by dmitry-timofeev** [comment]
Thanks, updated

File: src/get-started/java-binding.md (hunk 1 above)
Comment (Maria-Nosyk) [reply src/get-started/java-binding.md]: I would then put indefinite articles in the first sentence as  well:
>  The external service API is used for the interaction between a service and external systems.

File: src/get-started/java-binding.md (hunk 4 above)
Comment (dmitry-timofeev) [reply src/get-started/java-binding.md]: It doesn't :astonished: 

@slowli , is it an overly strict linter that forbids using `1.` as in Markdown to specify any item of a numbered list, or is it a renderer itself that cannot put the proper numbers on list items?

**Comment by dmitry-timofeev** [comment]
the linters are finally happy :tada:

File: src/get-started/java-binding.md [hunk 6]
```diff
@@ -143,71 +143,92 @@ rules – see the corresponding section of our [documentation][transactions].
...
-as an argument – a view that allows performing modifying operations. A service
-schema object can be used to access data collections of this service.
+the [`Transaction`][transaction] interface and defining transaction
+business logic. The interface implementations must define an execution
+rule for the transaction in `execute` method.
+
+The `Transaction#execute` method describes the operations that are applied to the
+current storage state when the transaction is executed. Exonum passes
+an [execution context][transaction-execution-context] as an argument,
+which allows to get a `Fork` – a view that allows tp perform modifying
```
[the remark this answers is not in the document]
Comment (dmitry-timofeev) [reply src/get-started/java-binding.md]: I see, allow smb to do sth, thanks!

**Pull request merged**